prime-router/metadata/fhir_mapping/hl7/datatypes/XTN/ContactPoint.yml

```diff
@@ -0,0 +1,301 @@
+      value: $contactPointLocal
+
+extension_5-local-number:
+  condition: $localNum NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    localNum: String, XTN.7
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "local-number"
+    valueString:
+      value: $localNum
+
+extension_6-phone-extension:
+  condition: $numExt NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    numExt: String, XTN.8
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "contactpoint-extension"
+    valueString:
+      value: $numExt
+
+extension_7-any-text:
+  condition: $anyText NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    anyText: String, XTN.9
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "any-text"
+    valueString:
+      type: STRING
+      valueOf: $anyText
+
+extension_8-telecom-use-code:
+  condition: $useCode NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    useCode: String, XTN.2
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "telecom-use-code"
+    valueString:
+      value: $useCode
+
+extension_9-hl7-use:
+  condition: $hl7Use NOT_NULL
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "hl7-use"
+    valueString:
+      value: $hl7Use
+
+extension_10-telephone-number:
+  condition: $telNum NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    telNum: String, XTN.1
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "telephone-number"
+    valueString:
+      value: $telNum
+
+extension_11-communication-address:
+  condition: $commAddress NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    commAddress: String, XTN.4
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "communication-address"
+    valueString:
+      value: $commAddress
+
+extension_12-unformatted-telephone-number:
+  condition: $unformattedTelNum NOT_NULL
+  generateList: true
+  expressionType: nested
+  vars:
+    unformattedTelNum: String, XTN.12
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "unformatted-telephone-number"
+    valueString:
+      value: $unformattedTelNum
+
+period:
+  generateList: true
+  expressionType: nested
+  vars:
+    start: XTN.13
+    end: XTN.14
+  expressionsMap:
+    start:
+      condition: $start NOT_NULL
+      type: STRING
+      valueOf: "GeneralUtils.dateTimeWithZoneId(start,ZONEID)"
+      expressionType: JEXL
+    end:
+      condition: $end NOT_NULL
+      type: STRING
+      valueOf: "GeneralUtils.dateTimeWithZoneId(end,ZONEID)"
+      expressionType: JEXL
+
+rank:
+  type: POSITIVEINT
+  valueOf: XTN.18
+  expressionType: HL7Spec
+
+extension_13-hl7v2Name:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: "hl7v2Name"
+    valueString:
+      type: STRING
+      valueOf: $hl7v2Name
```

Comment:
this extension is meant to identify the HL7 origin/destination. There are instances when multiple items are added to telecom but there's no way to know which is the next of kin phone vs patient phone etc

Reply:
yep, exactly. NK1.40 and NK1.41 is an example